lib/OpenQA/Schema/Result/Assets.pm

```diff
@@ -41,4 +46,113 @@ __PACKAGE__->add_unique_constraint([qw/type name/]);
 __PACKAGE__->has_many(jobs_assets => 'OpenQA::Schema::Result::JobsAssets', 'asset_id');
 __PACKAGE__->many_to_many(jobs => 'jobs_assets', 'job');
 
+
+sub _getDirSize {
+    my ($dir, $size)  = @_;
+    $size //= 0;
+
+    opendir(my $dh, $dir) || return 0;
+    for my $dirContent (grep(!/^\.\.?/,readdir($dh))) {
+
+        $dirContent = "$dir/$dirContent";
+
+        if (-f $dirContent) {
+            my $fsize = -s $dirContent;
+            $size += $fsize;
+        }
+        elsif (-d $dirContent) {
+            $size = _getDirSize($dirContent, $size);
+        }
+    }
+    closedir($dh);
+    return $size;
+}
+
+sub disk_file {
+    my ($self) = @_;
+    sprintf("%s/%s/%s", $OpenQA::Utils::assetdir, $self->type, $self->name);
+}
+
+sub remove_from_disk {
+    my ($self) = @_;
+
+    my $file = $self->disk_file;
+    OpenQA::Utils::log_debug("RM $file");
+    if ($self->type eq 'iso') {
+        unlink($file);
+    }
+    elsif ($self->type eq 'repo') {
+        use File::Path qw(remove_tree);
+        remove_tree($file);
+    }
+
+}
+
+sub ensure_size {
+    my ($self) = @_;
+
+    return $self->size if defined($self->size);
+
+    my $size = 0;
+    my @st = stat($self->disk_file);
+    if (@st) {
+        if ($self->type eq 'iso') {
+            $size = $st[7];
+        }
+        elsif ($self->type eq 'repo') {
+            $size = _getDirSize($self->disk_file);
+        }
+    }
+    $self->update({size => $size}) if $size;
+    return $size;
+}
+
+# this is a GRU task - abusing the namespace
+sub limit_assets {
+    my ($app) = @_;
+    my $groups = $app->db->resultset('JobGroups')->search({}, { select => 'id' });
```

Comment:
Should there be also 'size_limit_gb' in select? That I think is the culprit failing travis tests.